src/config.ts

```diff
@@ -1,14 +1,93 @@
+    : isLocal);
+
+export const appTitle = (isLocal ? local.appTitle : process.env.APPTITLE) || 'NORA';
+
+function fileOrEnv(key: string, keyFile: string) {
+	if (process.env[key]) {
+		return process.env[key];
+	}
+	if (typeof process.env[keyFile] === 'string') {
+		return fs.readFileSync(process.env[keyFile]);
+	}
+	return;
+}
+
+let tls = {
+	key: fileOrEnv('TLS_KEY', 'TLS_KEY_FILE'),
+	cert: fileOrEnv('TLS_CERT', 'TLS_CERT_FILE')
+};
+if (!(tls.key && tls.cert)) {
+	tls = undefined;
+}
+export const serviceSockets = isLocal ? local.serviceSockets : [
+	{
+		port: isLocal ? local.port : process.env.PORT,
+		address: isLocal ? local.address : process.env.ADDRESS,
+		tls
+	}
+];
 export const oauthClientId = isLocal ? local.oauthClientId : process.env.OAUTH_ID;
 export const oauthClientSecret = isLocal ? local.oauthClientSecret : process.env.OAUTH_SECRET;
 export const jwtCookieName = isLocal ? local.jwtCookieName : process.env.JWT_COOKIE;
-export const jwtSecret = isLocal ? local.jwtSecret : process.env.JWT_SECRET;
-export const projectId = isLocal ? local.projectId : process.env.PROJECT_ID;
 export const googleProjectApiKey = isLocal ? local.googleProjectApiKey : process.env.PROJECT_API_KEY;
-export const postgressConnectionString = isLocal ? local.postgressConnectionString : process.env.DATABASE_URL;
-export const serviceAccountIssuer = isLocal ? local.serviceAccountIssuer : process.env.SERVICE_ACCOUNT_ISSUER;
-export const serviceAccountPrivateKey = isLocal ? local.serviceAccountPrivateKey : process.env.SERVICE_ACCOUNT_KEY;
+export const serviceAccount = {
+    project_id: isLocal ? local.projectId : process.env.PROJECT_ID,
+    client_email: isLocal ? local.serviceAccountClientEmail : process.env.SERVICE_ACCOUNT_CLIENT_EMAIL,
+    private_key: isLocal ? local.serviceAccountPrivateKey : process.env.SERVICE_ACCOUNT_PRIVATE_KEY
+};
+const serviceAccountJsonFileName = isLocal ? local.serviceAccountJson : process.env.SERVICE_ACCOUNT_JSON;
+if (typeof serviceAccountJsonFileName === 'string') {
+    Object.assign(serviceAccount, JSON.parse(fs.readFileSync(serviceAccountJsonFileName).toString()));
+}
+export const oauthProjectId = isLocal ? local.projectId : process.env.OAUTH_PROJECT_ID || serviceAccount.project_id;
+export const jwtSecret = isLocal ? local.jwtSecret : process.env.JWT_SECRET || serviceAccount.private_key;
+
+export const noraServiceUrl = isLocal ? local.noraServiceUrl : process.env.NORA_SERVICE_URL || 'node-red';
+
+let ssl = true;
+if (isLocal && typeof local.postgresSsl === 'boolean') {
+    ssl = local.postgresSsl;
+} else if (typeof process.env.POSTGRES_SSL === 'string') {
+    const sslString = process.env.POSTGRES_SSL;
+    if (sslString.startsWith('F') || sslString.startsWith('f') || parseInt(sslString, 10) === 0) {
+        ssl = false;
+    }
+}
+export const postgres = {
+    connectionString: isLocal ? local.postgresConnectionString : process.env.POSTGRES_CONNECTIONSTRING,
```

Comment:
I think the problem comes from then env variables. Can you leave them with the name they had.

Reply:
Sorry for that, I had a bit lazy on the last changes. I now made these three env-vars:
DATABASE_URL
SERVICE_ACCOUNT_ISSUER
SERVICE_ACCOUNT_KEY
backward compatible. Hopefully I get all of the non compatible changes yet.

Reply:
I might be an good idea to transform the configure to an service. But on the other hand I need to make the @Get('/whatever') annotations working with an dynamic injection an Base director. This is somehow needed for the google cloud function runtime. What do you thing might be the best why. To make that dynamic this function like:
@Get('/whatever', cfg.baseDir) or to inject the config into the AnnotationService.

Reply:
Not sure I understand the context. Why do you need the config in `@get`?

Reply:
Basically I  will change the currently static config to an injected object.
To run on firebase the url locals like: 
https://firebase/base/.....
So your pattern for adding a route has do be configure dependable. Thats why I thought to added an parameter to cfg. 
So just don't care now, I will introduce next the configuration to be injectable.

Reply:
If you want to add a prefix (like `base` or `api`) to all the routes you can do it directly in `src/http/app.ts` https://github.com/andrei-tatar/nora-service/blob/master/src/http/app.ts#L21

Reply:
we need that information also in the redirects 302. But currently I'm fine with that.